### PR TITLE
Fix external VFS paths

### LINKED ISSUE #307: Using `apple_framework` to build an external framework fails: `<unknown>:0: error: underlying Objective-C module <module-name> not found`
When building an external dependency with `apple_framework`, I get the following error:
```
ERROR: /private/var/tmp/_bazel_jschear/7436e6242b3207e4692683d02ddfce6d/external/com_github_apple_swiftcollections/BUILD.bazel:5:16: Compiling Swift module DequeModule failed: (Exit 1): worker failed: error executing command
  (cd /private/var/tmp/_bazel_jschear/7436e6242b3207e4692683d02ddfce6d/execroot/build_bazel_rules_ios && \
  exec env - \
    APPLE_SDK_PLATFORM=iPhoneSimulator \
    APPLE_SDK_VERSION_OVERRIDE=14.5 \
    XCODE_VERSION_OVERRIDE=12.5.1.12E507 \
  bazel-out/host/bin/external/build_bazel_rules_swift/tools/worker/worker swiftc @bazel-out/ios-x86_64-min13.3-applebin_ios-ios_x86_64-dbg-ST-8633e27d3539/bin/external/com_github_apple_swiftcollections/DequeModule.swiftmodule-0.params)
Execution platform: @local_config_platform//:host
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
```

The build worked before this commit: https://github.com/bazel-ios/rules_ios/commit/c145ef3c1d5c5937195e6d7d40111db2640b197b

Here's the diff between `DequeModule_swift_vfs.yml` on that commit & the commit before it:
```diff
diff --git DequeModule_swift_vfs_old.yaml DequeModule_swift_vfs.yaml
index dc65ecb..96cfd61 100755
--- DequeModule_swift_vfs_old.yaml
+++ DequeModule_swift_vfs.yaml
@@ -1,6 +1,6 @@
 {
   "case-sensitive": true,
-  "overlay-relative": true,
+  "overlay-relative": false,
   "roots":
     [
       {
@@ -16,7 +16,7 @@
                   {
                     "type": "file",
                     "name": "DequeModule-umbrella.h",
-                    "external-contents": "../../../bazel-out/ios-x86_64-min13.3-applebin_ios-ios_x86_64-dbg-ST-8633e27d3539/bin/external/com_github_apple_swiftcollections/DequeModule-modulemap/DequeModule-umbrella.h",
+                    "external-contents": "bazel-out/ios-x86_64-min13.3-applebin_ios-ios_x86_64-dbg-ST-8633e27d3539/bin/external/com_github_apple_swiftcollections/DequeModule-modulemap/DequeModule-umbrella.h",
                   },
                 ],
             },
@@ -28,7 +28,7 @@
                   {
                     "type": "file",
                     "name": "module.modulemap",
-                    "external-contents": "../../../bazel-out/ios-x86_64-min13.3-applebin_ios-ios_x86_64-dbg-ST-8633e27d3539/bin/external/com_github_apple_swiftcollections/DequeModule-modulemap/DequeModule.modulemap",
+                    "external-contents": "bazel-out/ios-x86_64-min13.3-applebin_ios-ios_x86_64-dbg-ST-8633e27d3539/bin/external/com_github_apple_swiftcollections/DequeModule-modulemap/DequeModule.modulemap",
                   },
                 ],
             },
@@ -37,7 +37,7 @@
       {
         "type": "file",
         "name": "/build_bazel_rules_ios/frameworks/DequeModule.swiftmodule",
-        "external-contents": "../../../bazel-out/ios-x86_64-min13.3-applebin_ios-ios_x86_64-dbg-ST-8633e27d3539/bin//DequeModule.swiftmodule",
+        "external-contents": "bazel-out/ios-x86_64-min13.3-applebin_ios-ios_x86_64-dbg-ST-8633e27d3539/bin//DequeModule.swiftmodule",
       },
     ],
   "use-external-names": false,
```

I'll open a PR with a reproducer.

### PR DESCRIPTION
**Pull request by jerrymarino**
Fix external VFS paths

Currently, the VFS rule formulates a path relative to the BUILD file
path, to replicate the "exec path" in the java rules. Now this path
formulation include the repository name for external repos. Of course,
it isn't returned by `build_file_path`. We didn't have a test case and
thanks to @jschear, it's now up and running

Fixes #307

### CODE REVIEW

**Comment by tinder-maxwellelliott** [comment]
Thanks @jerrymarino , really appreciate the fix.

This seems to be working well for us, but I am seeing this error when compiling a genrule source

```
bazel-out/ios-x86_64-min12-applebin_ios-ios_x86_64-fastbuild-ST-e32b3366c2e3/bin/Projects/Modules/Utility/SomeModule/SomeModuleGeneratedMocks.swift:9:18: error: cannot load underlying module for 'SomeModule'
@testable import SomeModule
```

**Comment by jerrymarino** [comment]
@tinder-maxwellelliott cool, glad it fixes the external issue. Maybe I can throw a genrule source in there real quick today as well. Are you hitting that with or without "--features apple.virtualize_frameworks"?

I noticed one other issue that requires me to override `enable_framework_vfs` in the `apple_library`, which was a temp workaround that needs to be fixed.

**Comment by jerrymarino** [comment]
e.g. for all target types I end up setting `enable_framework_vfs=True` as an arg to all `apple_library` invocation

**Comment by tinder-maxwellelliott** [comment]
> @tinder-maxwellelliott cool, glad it fixes the external issue. Maybe I can throw a genrule source in there real quick today as well. Are you hitting that with or without "--features apple.virtualize_frameworks"?

Adding `enable_framework_vfs=True` to my `apple_library` targets got almost everything working, thanks 😉 

Now I am hitting and issue with `apple_static_framework_import`, getting `error: missing required modules` on those

**Comment by tinder-maxwellelliott** [comment]
Update: I am able to run my main app target with this PR, but I am unable to run its tests, still seeing `missing required modules` error. Ill work on a reproducer

**Comment by jerrymarino** [comment]
@tinder-maxwellelliott awesome that's great to hear! Yeah if you can come up with a reproducer, we can make sure it's working and doesn't break again for the use case you had in mind going forward 👍 .

File: rules/framework/vfs_overlay.bzl [hunk 1]
```diff
@@ -28,7 +28,12 @@ def _get_external_contents(prefix, path_str):
     return prefix + path_str
 
 def _get_vfs_parent(ctx):
-    return (ctx.bin_dir.path + "/" + ctx.build_file_path)
+    root_path = ctx.bin_dir.path + "/"
+    # For an external package, the BUILD file will be rooted relative to the
+    # workspace_root, account for this
+    if len(ctx.label.workspace_root):
+        root_path += ctx.label.workspace_root + "/"
```
Comment (gyfelton) [inline rules/framework/vfs_overlay.bzl:35]: buildifier seems to want to have a empty line after this line

File: WORKSPACE [hunk 2]
```diff
@@ -82,3 +82,10 @@ http_file(
     # .bazelversion uses 4.0
     urls = ["https://github.com/bazelbuild/buildtools/releases/download/3.5.0/buildifier.mac"],
 )
+
+load(
```
Comment (congt) [inline WORKSPACE:86]: nit: Should we add a comment here?

**Review by congt — APPROVED** [review]

File: WORKSPACE (hunk 2 above)
Comment (jerrymarino) [reply WORKSPACE:86]: I think the name is pretty self documenting, but not opposed to a comment  `load_external_test_dependency`

**Review by thiagohmcruz — APPROVED** [review]

**Pull request merged**